server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStoreTests.java

```diff
@@ -332,6 +337,32 @@ void beOAuth2StandardCompliant() {
         assertTrue(code.length() >= 32);
     }
 
+    @Test
+    void testCountingTheExecutedSqlDeleteStatements() throws SQLException {
+        // Given, mocked data source to count how often it is used, call performExpirationClean 10 times
+        DataSource mockedDataSource = mock(DataSource.class);
+        Connection mockedConnection = mock(Connection.class);
+        doReturn(mockedConnection).when(mockedDataSource).getConnection();
+        store = new UaaTokenStore(mockedDataSource);
```

Comment:
I spent 15 minutes staring at this code to understand what it does, but I just can't. I don't understand why you are mocking the `getConnection()` method. I can't even see it used in the `UaaTokenStore` code. 

It looks to me like you are doing some quite low level white box testing. I would much rather try to write high level tests.